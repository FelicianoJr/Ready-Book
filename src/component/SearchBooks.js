import React from 'react'
import { Link } from 'react-router-dom'
import { Debounce } from 'react-throttle';
import BookDetail from './BookDetail'

const SearchBooks = ({ bookSearch, onSearchBook, update }) => {

    return (
        <div>
            <div className="search-books">
                <div className="search-books-bar">
                    <Link className="close-search" to="/"></Link>
                    <div className="search-books-input-wrapper">
                        <Debounce time="400" handler="onChange">
                            <input
                                type='text'
                                placeholder='search by title or author'
                                onChange={event => onSearchBook(event.target.value)}
                            />
                        </Debounce>
                    </div>
                </div>
                <div className="search-books-results">
                    <ol className="books-grid">
                        {bookSearch.map(book => {
                            return <BookDetail key={book.id} book={book} update={update} />
                        })}
                    </ol>
                </div>
            </div>
        </div>
    )
};

export default SearchBooks;