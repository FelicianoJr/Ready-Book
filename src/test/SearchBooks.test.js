import SearchBooks from '../component/SearchBooks'
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("<SearchBooks/>", () => {

    let update;
    let props;
    let wrapper;

    beforeEach(() => {
        update = jest.fn();
        props = {
            bookSearch: [{
                id: 0, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React', shelf: 'read'
            },
            {
                id: 1, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React Native', shelf: 'read'
            }],
            onSearchBook: update,
        };
        wrapper = shallow(<SearchBooks  {...props} />)
    });

    it('should render input equal one', () => {
        expect(wrapper.find('input').length).toBe(1);
    });

    it("responds to name change", () => {
        wrapper.find('input').simulate('change', { target: { value: 'artificial' } });
        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('should render one <ol/>', () => {
        expect(wrapper.find('ol').length).toEqual(1);
    });

})