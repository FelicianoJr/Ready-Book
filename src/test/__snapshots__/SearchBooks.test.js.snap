exports[`<SearchBooks/> responds to name change 1`] = `
<div>
  <div
    className="search-books">
    <div
      className="search-books-bar">
      <Link
        className="close-search"
        replace={false}
        to="/" />
      <div
        className="search-books-input-wrapper">
        <Debounce
          handler="onChange"
          handlers={Array []}
          time="400">
          <input
            onChange={[Function]}
            placeholder="search by title or author"
            type="text" />
        </Debounce>
      </div>
    </div>
    <div
      className="search-books-results">
      <ol
        className="books-grid">
        <BookDetail
          book={
            Object {
              "id": 0,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React",
            }
          }
          key="0" />
        <BookDetail
          book={
            Object {
              "id": 1,
              "imageLinks": Object {
                "thumbnail": "url/test/enzime",
              },
              "shelf": "read",
              "title": "React Native",
            }
          }
          key="1" />
      </ol>
    </div>
  </div>
</div>
`;
